hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAHSv2ClientImpl.java

```diff
@@ -107,6 +110,45 @@ public void testGetAppAttemptReport() throws IOException, YarnException {
         isEqualTo("test original tracking url");
   }
 
+  @Test
+  public void testGetContainerByAppAttempt() throws IOException, YarnException {
+    final ApplicationId appId1 = ApplicationId.newInstance(0, 1);
+    final ApplicationId appId2 = ApplicationId.newInstance(0, 2);
+
+    final ApplicationAttemptId appAttemptId1 =
+            ApplicationAttemptId.newInstance(appId1, 1);
+    final ApplicationAttemptId appAttemptId2 =
+            ApplicationAttemptId.newInstance(appId2, 2);
+
+    final ContainerId containerId1 = ContainerId.newContainerId(appAttemptId1, 1);
+    final ContainerId containerId2 = ContainerId.newContainerId(appAttemptId2, 2);
+
+
+    Map<String, String> filter1 = new HashMap<>();
+    filter1.put("appattemptId", appAttemptId1.toString());
+    Map<String, String> filter2 = new HashMap<>();
+    filter2.put("appattemptId", appAttemptId2.toString());
```

Comment:
Nit: Maybe an inline ImmutableMap.of would be a bit cleaner, because the filter maps aren't reused.